Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,208 @@
+//-----------------------------------------------------------------------
+// <copyright file="TreeSchemaValidatorTests.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     Tests for the TreeSchemaValidator class.
+// </summary>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Forge.TreeWalker.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.Forge.DataContracts;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Schema;
+    using Microsoft.Forge.TreeWalker;
+
+    [TestClass]
+    public class TreeSchemaValidatorTests
+    {
+        private string tardigradeSchemaString;
+        private ForgeTree treeSchema;
+        private Dictionary<string, ForgeTree> treeSchemas;
+        private string tardigradeSchemaPath;
+        private string schemaDirectoryPath;
+        private string stringRules;
+        private string rulesForDictionary;
+        private JSchema jschemaRules;
+        private string invalidSchemaDirectoryPath;
+        private string invalidSchemaWithErrorContent;
+        private string subroutineSchemaString;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            tardigradeSchemaPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\TardigradeSchema.json");
+            schemaDirectoryPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas");
+            stringRules = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaValidationRules.json"));
+            rulesForDictionary = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaDictionaryValidationRules.json"));
+            jschemaRules = JSchema.Parse(stringRules);
+            tardigradeSchemaString = File.ReadAllText(tardigradeSchemaPath);
+            subroutineSchemaString = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\SubroutineSchema.json"));
+            treeSchema =  JsonConvert.DeserializeObject<ForgeTree>((string)tardigradeSchemaString);
+            treeSchemas = new Dictionary<string, ForgeTree>();
+            treeSchemas.Add("tree1", treeSchema);
+            treeSchemas.Add("tree2", treeSchema);
+            treeSchemas.Add("tree3", treeSchema);
+            invalidSchemaWithErrorContent = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\InvalidTestSchemas\\InvalidTestSchemaErrorContent.json"));
+            invalidSchemaDirectoryPath = "test\\ExampleSchemas\\TardigradeSchema.json";
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInForgeTreeWithCustomRulesInString()
```

Comment:
>Test_ValidateSchemaInForgeTreeWithCustomRulesInString [](http://example.com/codeflow?start=20&length=53)

Let's have the test method names also align more with the ForgeSchemaValidator methods.

e.g.
Test_ValidateSchemaAsForgeTree_WithRulesAsString
Test_ValidateSchemaAsForgeTree_WithRulesAsJSchema
Test_ValidateSchemaAsForgeTreeDictionary_WithValidateAsDictionary
Test_ValidateSchemaAsForgeTreeDictionary_WithoutValidateAsDictionary
Test_ValidateSchemaAsString_WithForgeTree